example/example.client.ts

```diff
@@ -422,29 +422,49 @@ export type Implementation = (
 
 export class ExpressZodAPIClient {
   constructor(protected readonly implementation: Implementation) {}
+  private parseRequest(request: string) {
+    return request.split(/ (.+)/, 2) as [Method, Path];
+  }
+  private substitute(path: string, params: Record<string, any>) {
+    const rest = { ...params };
+    for (const key in params) {
+      path = path.replace(`:${key}`, () => {
+        delete rest[key];
+        return params[key];
+      });
+    }
+    return [path, rest] as const;
+  }
   public provide<K extends Request>(
     request: K,
     params: Input[K],
   ): Promise<Response[K]> {
-    const [method, path] = request.split(/ (.+)/, 2) as [Method, Path];
-    return this.implementation(
-      method,
-      Object.keys(params).reduce(
-        (acc, key) =>
-          acc.replace(`:${key}`, (params as Record<string, any>)[key]),
-        path,
-      ),
-      Object.keys(params).reduce(
-        (acc, key) =>
-          Object.assign(
-            acc,
-            !path.includes(`:${key}`) && {
-              [key]: (params as Record<string, any>)[key],
-            },
-          ),
-        {},
-      ),
+    const [method, path] = this.parseRequest(request);
+    return this.implementation(method, ...this.substitute(path, params));
+  }
+  public subscribe<
```

Comment:
maybe this should not be within client, because the client instantiation requires `Implementation` only needed for `provide()`, while `subscribe()` does not use it.
Perhaps `parseRequest()`, `substitute()` and `subscribe()` should be just functions.

Reply:
fixed in f66e1175